cql3/expr/expression.hh

```diff
@@ -389,6 +428,8 @@ struct usertype_constructor {
     // Might be nullptr before prepare.
     // After prepare always holds a valid type, although it might be reversed_type(user_type).
     data_type type;
+
+    friend bool operator==(const usertype_constructor&, const usertype_constructor&) = default;
 };
 
 // now that all expression types are fully defined, we can define expression::impl
```

Comment:
By using `operator==() = default` we often end up comparing `shared_ptr` instead of their contents.
Sometimes it would be easy to compare the contents, e.g. in `unresolved_identifier`. It would be a more useful operator if it could handle such differences, but I guess it matches the definition given in the comment.

Reply:
`data_type` does interning, so usually comparing the pointer is just fine.